core/modules/query/ValueExpr.cc

```diff
@@ -176,11 +198,10 @@ double ValueExpr::copyAsType<double>(double const& defaultValue) const;
 template int ValueExpr::copyAsType<int>(int const&) const;
 
 
-void ValueExpr::findColumnRefs(ColumnRef::Vector& vector) {
-    for(FactorOpVector::iterator i=_factorOps.begin();
-        i != _factorOps.end(); ++i) {
-        assert(i->factor); // FactorOps should never have null ValueFactors
-        i->factor->findColumnRefs(vector);
+void ValueExpr::findColumnRefs(ColumnRef::Vector& vector) const {
+    for (auto factorOp : _factorOps){
```

Comment:
`auto&&`?

Reply:
probably `const auto&` to force preserve constness?